staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -721,17 +721,17 @@ func shouldDelegateList(opts storage.ListOptions) bool {
 	pred := opts.Predicate
 	match := opts.ResourceVersionMatch
 	pagingEnabled := utilfeature.DefaultFeatureGate.Enabled(features.APIListChunking)
+
+	// Serve consistent reads from storage
+	consistentReadFromStorage := resourceVersion == ""
+	// Watch cache doesn't support continuations, so serve them from etcd.
 	hasContinuation := pagingEnabled && len(pred.Continue) > 0
+	// Serve paginated requests about revision "0" from watch cache to avoid overwhelming etcd.
 	hasLimit := pagingEnabled && pred.Limit > 0 && resourceVersion != "0"
+	// Watch cache only supports ResourceVersionMatchExact (default).
```

Comment:
It's vice versa - it doesn't support ResourceVersionMatchExact.

Reply:
Ups, would be great to remove negations to make it easier to comprehend. With the new tests it should be safe.